### DISCUSSION TITLE
Tooltip tidak muncul dan warna bar juga tidak bisa berubah

### DISCUSSION THREAD

**Discussion by arip7608** [teamalgoritma/community#883] [Q&A] [answered]
dear team algorithma , help please
di code bawah ini saya membuat data visualisasi dimana saya mendapatkan beberapa error sebagai berikut 
1.  Tooltip tidak muncul dan coba memberikan syntax comm(), di glue tapi tidak keluar.
2. warna di bagian bar plot nya tidak berubah (hanya warna grey.)
3. Ingin memberikan pemisah ribuan di bagian sumbu x bagaimana ya.

mohon koreksinya atas kode dibawah ini.

```{r}
online1 <- online %>% 
  filter(Month %in% "Jan") %>% 
  group_by(Product.Category) %>% 
  summarise(to_rev = sum(Total.Revenue)) %>% 
  arrange(-to_rev) %>% 
  mutate(label = glue("Revenue of : {Product.Category}
                      comma ({to_rev})"))
```
```{r}
 plot1 <- 
  ggplot(data = online1, mapping = aes(x = to_rev, 
                                       y = reorder (Product.Category,to_rev)), 
         fill = "blue", 
         text = label) + 
  geom_col() + 
  scale_fill_gradient( low = "blue",
                       high = "red") +
  labs(title = "Revenue Contribution by Product Category",
        x = "Revenue",
       y = " Product Category") +
  theme_classic()+
  theme(legend.position = "none")
  

```
```{r}
ggplotly(plot1, tooltip = "text")
```
![image](https://github.com/teamalgoritma/community/assets/173215925/68d02571-3a6d-46da-9143-6d2931da1f10)

**Comment by ramdhinafinita** [comment, marked as answer]
Dear @arip7608 

1. Tooltip tidak muncul dikarenakan penempatan parameter text = label seharusnya berada di dalam aes( ) pada ggplot berbarengan setelah  y = reorder (Product.Category,to_rev), text = label) 
2. Warna di bagian bar plot nya tidak berubah (hanya warna grey), hal ini dikarenakan penempatan aes yang salah, seharusnya:
```
ggplot(mapping = aes(x=to_rev,
                     y = reorder (Product.Categoryto_rev, to_rev),
                     text = label))+
  geom_col(aes(fill = to_rev))+ #disesuaikan dengan apa yang mau digradasikan, nilai dari to_rev nya
  scale_fill_gradient( low = "blue",
                       high = "red") +
  labs(title = "Revenue Contribution by Product Category",
        x = "Revenue",
       y = " Product Category") +
  theme_classic()+
  theme(legend.position = "none")
```

3. Untuk pemisah koma, karena berada di sumbu x maka mengugnakan `scale_x_continuous(labels = comma) +`

Terimakasih, semoga membantu

**Comment by arip7608** [comment]
Terima kasih
